### PR TITLE
Packaging, Export Error Handling, Ignition Target & Type Dropdowns

### PR DESCRIPTION
**Pull request by mdbernard**

### CODE REVIEW

**Comment by bryanwweber** [comment]
By the way @mdbernard Can you put the issue numbers that this fixes in a comment, so they get automatically closed when this is merged? Something like "Fixes #X, #Y, #Z"

**Comment by mdbernard** [comment]
Fixes #7, #8, #9

File: chemkedGui/chemkedGui.py [hunk 1]
```diff
@@ -459,68 +459,71 @@ def export(self):
...
-                    'pages': self.file['reference']['pages'].text(),
-                    'detail': self.file['reference']['detail'].text()
-                },
-                'apparatus': {
-                    'kind': self.file['apparatus']['kind'].currentText(),
-                    'institution': self.file['apparatus']['institution'].text(),
-                    'facility': self.file['apparatus']['facility'].text(),
-                },
-                'datapoints': datapoints
+        # if save_location[0] != '':
```
Comment (bryanwweber) [inline chemkedGui/chemkedGui.py]: Better not to commit commented-out code 😄 Can you remove this line?

**Comment by mdbernard** [comment]
Hello Prof.,

Here are the changes to the currently open PR since our meeting:

   - Removed that unnecessary commented line you pointed out in the export
   method.
   - Renamed lots of variables and methods to snake_case to match PEP8.
   - Renamed the directory and module from chemkedGui to chemked_gui to
   match PEP8.
   - Moved lots of code from Contents.__init__() to specific setup methods
   for each tab for better readability (tab_meta_setup, etc.).
   - Added CHANGELOG.md file and filled in basic descriptions of 0.1.0,
   0.1.1, and 0.1.2.

The skip_validation is still set to True. I'll try to take a look at where
it's going wrong when changing that to False, but I probably won't get to
it until after finals. Let me know if there's anything else that ought to
be changed.

Also, I don't remember if I ever let you know that I filled in the Wiki a
bit. Right now it just explains how to use the GUI, which is hopefully
self-explanatory. Would this be better used as a place to put more in-depth
documentation of the code rather than in the comments?

Mike

**Comment by bryanwweber** [comment]
@mdbernard The ideal place for documentation is in the code itself, and even better is docstrings for the various classes and functions. You can follow the "Google docstring style": http://www.sphinx-doc.org/en/master/ext/example_google.html#example-google which is also used by PyKED. I think the wiki is a good spot for documenting how to use the GUI though.

Do you want me to merge this as-is or do you want to fix the skip_validation here?

**Comment by mdbernard** [comment]
I would merge as-is for now. Users shouldn't have a problem (famous last
words). I'll make a PR with the skip_validation next.

**Pull request merged**